### DISCUSSION TITLE
imodeljs-native@2.13.3 integrity error

### DISCUSSION THREAD

**Discussion by DeepakMalkan** [iTwin/itwinjs-core#1694] [Q&A] [answered]
I'm creating an agent service: https://www.itwinjs.org/learning/tutorials/develop-agent/

After cloning the git repository, I get the following error message on 'npm install':

$ npm install
npm WARN tarball tarball data for @bentley/imodeljs-native@2.13.3 (sha512-CMSmGw9buvYfaxSTDBIlfVK6jM8DrgpjsvDyFL1fu7j+BXw0Y6vCGNnPFa7t3gtE92uo9qzmKHwBOBvHFXujHw==) seems to be corrupted. Trying one more time.
npm ERR! code EINTEGRITY
npm ERR! sha512-CMSmGw9buvYfaxSTDBIlfVK6jM8DrgpjsvDyFL1fu7j+BXw0Y6vCGNnPFa7t3gtE92uo9qzmKHwBOBvHFXujHw== integrity checksum failed when using sha512: wanted sha512-CMSmGw9buvYfaxSTDBIlfVK6jM8DrgpjsvDyFL1fu7j+BXw0Y6vCGNnPFa7t3gtE92uo9qzmKHwBOBvHFXujHw== but got sha512-iYzSerUJytkOqoc93ECMIXg3rYcOTbNexZnp2llmtHH2b1p4AXlIjDjUKVDfXlNP0AKKtUy1q/6yow//NcSpAg==. (35314 bytes)

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Deepak.Malkan\AppData\Roaming\npm-cache\_logs\2021-06-22T19_20_45_654Z-debug.log

How can I fix this problem?

Deepak.

**Comment by calebmshafer** [comment, marked as answer]
@DeepakMalkan this happens when you're switching between the npmjs registry and another registry such as Azure Artifacts.  The easiest way to fix it is to delete the `package-lock.json` and run it.  I'll see what we can do about making it so you won't run into this in either case.

**Reply by DeepakMalkan** [reply]
Thx. Deleting package-lock.json worked.
